### PR TITLE
arangodb: cleanup starter installation

### PR DESCRIPTION
**Pull request by bayandin**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This PR cleans up installation of a starter of arangodb:
- adds a tag to starter resource 
- deprecates GOPATH
- uses `std_go_args`

### CODE REVIEW

File: Formula/arangodb.rb [hunk 1]
```diff
@@ -22,22 +23,24 @@ class Arangodb < Formula
...
-      system "go", "build", "-ldflags", "-X main.projectVersion=0.14.15 -X main.projectBuild=#{commit}",
-                            "-o", "arangodb",
-                            "github.com/arangodb-helper/arangodb"
-      bin.install "arangodb"
+      ldflags = %W[
+        -s -w
+        -X main.projectVersion=#{resource("starter").version}
+        -X main.projectBuild=#{commit}
+      ]
+      system "go", "build", *std_go_args, "-ldflags", ldflags.join(" "), "github.com/arangodb-helper/arangodb"
```
Comment (dawidd6) [inline Formula/arangodb.rb:43]: I would try:

```suggestion
      system "go", "build", *std_go_args, "-ldflags", ldflags.join(" ")
```

Comment (bayandin) [reply Formula/arangodb.rb:44]: Nah, since the formula is using `go@1.13` we need explicitly set `GO111MODULE=on`.
Anyway, thanks for looking at it ❤️

**Review by chenrui333 — APPROVED** [review]
💯

File: Formula/arangodb.rb [hunk 2]
```diff
@@ -22,22 +23,25 @@ class Arangodb < Formula
...
+      tag:      "0.14.15",
       revision: "e32307e9ae5a0046214cb066355a8577e6fc4148"
   end
 
   def install
     ENV["MACOSX_DEPLOYMENT_TARGET"] = MacOS.version
 
     resource("starter").stage do
-      ENV["GOPATH"] = Dir.pwd + "/.gobuild"
+      ENV["GO111MODULE"] = "on"
```
Comment (chenrui333) [inline Formula/arangodb.rb:34]: technically, you dont need this. But I guess there is no harm though. :)

**Pull request closed without merging**